sky/utils/controller_utils.py

```diff
@@ -969,7 +969,9 @@ def _sub_path_join(sub_path: Optional[str], path: str) -> str:
     copy_mounts_with_file_in_src = {}
     for i, (dst, src) in enumerate(copy_mounts.items()):
         assert task.file_mounts is not None
-        task.file_mounts.pop(dst)
+        file_mounts = task.file_mounts
+        file_mounts.pop(dst)
+        task.set_file_mounts(file_mounts)
```

Comment:
This will cause the `_file_mounts` to be replaced wiht the substitute version right? Should we directly pop it from `_file_mounts` instead?